frontend/public/components/catalog/catalog-page.tsx

```diff
@@ -47,328 +47,335 @@ import { Item } from './types';
 export const CatalogListPage = withExtensions<CatalogListPageExtensionProps>({
   devCatalogExtensions: isDevCatalogModel,
 })(
-  class CatalogListPage extends React.Component<CatalogListPageProps, CatalogListPageState> {
-    constructor(props: CatalogListPageProps) {
-      super(props);
+  withQueryParams<CatalogListPageProps>(
+    class CatalogListPage extends React.Component<CatalogListPageProps, CatalogListPageState> {
+      constructor(props: CatalogListPageProps) {
+        super(props);
```

Comment:
Would be better just to convert this to a Functional Component.

Reply:
Catalog page is already getting a big refactor in other epic. So didn't wanted to make any big changes here.